extensions/EXT/EGL_EXT_query_reset_notification_strategy.txt

```diff
@@ -0,0 +1,78 @@
+Name
+
+	EXT_query_reset_notification_strategy
+
+Name Strings
+
+	EGL_EXT_query_reset_notification_strategy
+
+Contributors
+
+	Simon Zeni <simon.zeni@collabora.com>
+
+Contact
+
+	Simon Zeni <simon.zeni@collabora.com>
+
+Status
+
+	Draft
+
+Version
+
+	Version 0.1, 2023-11-01
+
+Number
+
+	EGL Extension #152
+
+Extension Type
+
+	Client
+
+Dependencies
+
+	Requiers EGL 1.4 and EXT_create_context_robustness.
+
+	The extension is written against the wording of the EGL 1.5
+	specification, but may be implemented against 1.4.
+
+Overview
+
+	This extension allows the user to query the reset notification strategy
+	of a given context using the function eglQueryContext.
+
+New Types
+
+	None
+
+New Procedures and Functions
+
+	None
+
+New Tokens
+
+	None
+
+Specification modifications
+
+	Append in section 3.7.4 (Context Queries), to the list of attributes
+	supported by eglQueryContext.
+
+		Querying EGL_CONTEXT_OPENGL_RESET_NOTIFICATION_STRATEGY_EXT
+		returns the reset notification strategy that was set for this
+		context. If no strategy was set, EGL_NO_RESET_NOTIFICATION_EXT
+		must be returned.
```

Comment:
What if it's not a robust context at all?  Is that intended to be the same as "no strategy set"?

Reply:
Mesa defaults it to `EGL_NO_RESET_NOTIFICATION_KHR` regardless of a robust context or not[1]

[1]: https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/egl/main/eglcontext.c?ref_type=heads#L615

Reply:
Probably, although this extension says it requires EXT_create_context_robustness, so I'm not sure how you'd get into that situation..

Reply:
With the requirement, I think it's implicit that the query returns `EGL_BAD_ATTRIBUTE` if `EXT_create_context_robustness` is not supported. 

Should it still return `EGL_NO_RESET_NOTIFICATION_KHR` if the context is not robust? Or should we introduce a new value to be returned? I quickly looked at `GL_EXT_robustness` and I don't see a different behaviour if the context is not robust.

Reply:
I guess it's fine then. I haven't thought about EGL much in a few years now! ;-)